include/openthread/platform/psa.h

```diff
@@ -0,0 +1,237 @@
+ * @retval OT_ERROR_INVALID_ARGS  @p aBuffer was NULL
+ *
+ */
+otError otPlatPsaExportKey(psa_key_id_t aKeyId, uint8_t *aBuffer, uint8_t aBufferLen, size_t *aKeyLen);
+
+/**
+ * Destroy a key stored in PSA ITS.
+ *
+ * @param[in]   aKeyId            Reference to the key to be used for crypto operations.
+ *
+ * @retval OT_ERROR_NONE          Successfully encrypted  @p aInput.
+ * @retval OT_ERROR_FAILED        Failed to encrypt @p aInput.
+ *
+ */
+otError otPlatPsaDestroyKey(psa_key_id_t aKeyId);
+
+/**
+ * Export public key of the keypair stored in ITS.
+ *
+ * @param[in]   aKeyId            Reference to the key to be exported.
+ * @param[out]  aOutput           Pointer to the buffer to store exported public key.
+ * @param[in]   aOutputSize       Size of the output buffer passed.
+ * @param[out]  aOutputLen        Length of the exported key.
+ *
+ * @retval OT_ERROR_NONE          Successfully exported  @p aKeyId.
+ * @retval OT_ERROR_FAILED        Failed to export @p aKeyId.
+ * @retval OT_ERROR_INVALID_ARGS  @p aOutput or @p aOutputLen was NULL
+ *
+ */
+otError otPlatPsaExportPublicKey(psa_key_id_t aKeyId, uint8_t *aOutput, size_t aOutputSize, size_t *aOutputLen);
+
+/**
+ * Sign a message hash using a key stored.
+ *
+ * @param[in]   aKeyId            Reference to the key to be used for Signing.
+ * @param[in]   aKeyAlgorithm     Key algorithm compatible with the key stored.
+ * @param[in]   aHash             Hash of the message to be signed.
+ * @param[in]   aHashSize         Size of the hash.
+ * @param[out]  aSignature        Buffer to store the calculated signature.
+ * @param[in]   aSignatureSize    Size of the buffer passed.
+ * @param[in]   aSignatureLen     Length of the signature calculated.
+ *
+ * @retval OT_ERROR_NONE          Successfully signed  @p aHash.
+ * @retval OT_ERROR_FAILED        Failed to sign @p aHash.
+ *
+ */
+otError otPlatPsaSignHash(psa_key_id_t    aKeyId,
+                          psa_algorithm_t aKeyAlgorithm,
+                          uint8_t *       aHash,
+                          size_t          aHashSize,
+                          uint8_t *       aSignature,
+                          size_t          aSignatureSize,
+                          size_t *        aSignatureLen);
+
+/**
+ * Verify a signature using a key stored.
+ *
+ * @param[in]   aKeyId            Reference to the key to be used for Verifying the signature.
+ * @param[in]   aKeyAlgorithm     Key algorithm compatible with the key stored.
+ * @param[in]   aHash             Hash of the message to be verified.
+ * @param[in]   aHashSize         Size of the hash.
+ * @param[out]  aSignature        Signature that needs verification.
+ * @param[in]   aSignatureSize    Size of the signature passed.
+ *
+ * @retval OT_ERROR_NONE          Successfully signed  @p aHash.
+ * @retval OT_ERROR_FAILED        Failed to sign @p aHash.
+ *
+ */
+otError otPlatPsaVerifyHash(psa_key_id_t    aKeyId,
```

Comment:
This seems to be unused. Do we need it, since we have equivalent PSA API?

Reply:
Yes, I will remove it from the commit.